Introduction/README.md

```diff
@@ -21,16 +21,68 @@ machine as Fn provides the necessary Go compiler and tools as a Docker
 container.  Let's walk through your first function to become familiar with the
 process and how Fn supports development.
 
-Before we start developing we need to set the `FN_REGISTRY`
-environment variable.  Normally, it's set to your Docker Hub username.
-However in this tutorial we'll work in local development mode so we can set
-the `FN_REGISTRY` variable to an arbitrary value. Let's use `fndemouser`.
+### Configure your Context
+Before we start developing we need to configure Fn to use a Docker registry.
+Normally, it's set to your Docker Hub username. However in this tutorial we'll
+work in a local development mode, so we will use an arbitrary value
+`fndemouser`. We store the registry value in an Fn context. An Fn context
+represents our current deployment environment and we can have more than one if
+we are deploying to multiple servers.
 
-![User Input Icon](images/userinput.png)
+First, get list of available contexts.
+
+![user input](images/userinput.png)
 >```sh
-> export FN_REGISTRY=fndemouser
+> fn list contexts
 >```
 
+The result should be similar to this:
+
+```txt
+CURRENT     NAME    PROVIDER    API URL                    REGISTRY
+            default default     http://localhost:8080/v1   
+```
+
+Notice we have a default context which deploys to a local Fn server. The default context is created the first time you run the Fn CLI. However, we need to select default as our current context and set a registry value to `fndemouser`.
+
+To do that we issue two commands. First select a context:
+
+![user input](images/userinput.png)
+>```sh
+> fn use context default
+>```
+
+```txt
+Now using context: default
+```
+
+Next, set the Docker registry value:
+
+![user input](images/userinput.png)
+>```
+> fn update context registry fndemouser
+>```
+
+```txt
+Current context updated registry with fndemouser
+```
+
+Now, recheck your context configuration:
+
+![user input](images/userinput.png)
+>```
+> fn list contexts
+>```
+
+```txt
+CURRENT     NAME    PROVIDER    API URL                    REGISTRY
+*           default default     http://localhost:8080/v1   fndemouser
+```
+
+The default context is now our current context and has a registry value of `fndemouser`. You are ready to create your first function.
+
```

Comment:
More generally, I feel this whole section on contexts is a big amount of duplication again and wonder whether it should form part of the overall initial install and config page, rather than being duplicated within each tutorial?